### ISSUE TITLE
Can't create unix socket without 777 permission

### ISSUE THREAD

**Issue by scaledteam** [w23/obs-kmsgrab#11]
Can't create unix socket without 777 permission in `/usr/local/share/obs/obs-plugins/linux-kmsgrab/` folder.

**Comment by w23** [comment]
Ah yes, we need to figure out a better path for socket.

**Comment by scaledteam** [comment]
obs-vkcapture uses this path "/tmp/obs-vkcapture.sock", and author also mentioned "$XDG_RUNTIME_DIR/obs-vkcapture.socket". But i don't sure about better socket location too, i'm new in unix sockets.

https://github.com/nowrep/obs-vkcapture/blob/3806d838a629884203c871602a817b2d9b8699ab/src/vkcapture.c#L76
https://github.com/nowrep/obs-vkcapture/issues/13#issuecomment-860911846

**Comment by w23** [comment]
> But i don't sure about better socket location too, i'm new in unix sockets.

Me too :D

Something something XDG, freedesktop docs, etc.
It should be a secure location only visible to current user, as this is a fairly security sensitive thing.

**Comment by adryd325** [comment]
seems like most other user specific sockets are in /run/user/1000 which is $XDG_RUNTIME_DIR

**Comment by xnqs** [comment]
Yeah, my fork actually does that and it works really well. Do note, though, that it uses an older, working version of the plugin that has implicit EGL dependency, and doesn't use the built-in OBS implementation.
